tests/PHPUnit/Unit/DataTable/Renderer/CSVTest.php

```diff
@@ -464,6 +464,42 @@ public function testRenderArray4()
         $this->assertEquals($expected, $render->render());
     }
 
+    /**
+     * @dataProvider getFormulaExpressions
+     */
+    public function testRendersFormulasAndNullBytesCorrectly($input, $expectedOutput)
+    {
+        $render = new Csv();
+        $render->setTable($input);
+        $render->convertToUnicode = false;
+        $expected = $expectedOutput;
+
+        $this->assertEquals($expected, $render->render());
+    }
+
+    public function getFormulaExpressions(): iterable
+    {
+        yield "formula starting with =, should be escaped with leading '" => [
+            ['=SUM(A)' => '=SUM(A;B)'], "'=SUM(A)\n\"'=SUM(A;B)\""
```

Comment:
Do we also need to backport #21719 to have this work fully as expected?

Reply:
makes sense. I'll add it to this PR